File: ClientApp/src/app/models/etf-country-allocation.ts
import { Country } from "./country";
import { Etf } from "./etf";
export class EtfCountryAllocation{
    etf: Etf;
    country: Country;
    allocation: number;
    public etfid: string;
    public countryid: string;

    constructor(etf: Etf, country: Country, allocation: number = 0){
        this.etf = etf;
        this.country = country;
        this.allocation = allocation;
    }
}

File: Models/Data/EtfCountryAllocation.cs
using System;

namespace MyEtf.Models.Data
{
    public class EtfCountryAllocation
    {
        public Guid Id {get;set;}
        public Guid EtfId {get;set;}
        public Guid CountryId {get;set;}
        public float Allocation {get;set;}

        public virtual Etf Etf {get;set;}
        public virtual Country Country {get;set;}
    }
}